app/controllers/application_controller.rb

```diff
@@ -108,7 +108,9 @@ def corporate_categories
   def corporate_category?(category, corporate = corporate_categories, categories = [])
     categories << corporate.map {|c| c['id']}
     return true if categories.flatten.include?(category)
-    corporate_category?(category, corporate.first['contents'], categories.flatten) if corporate.first['contents']
+    unless corporate.first['contents']
```

Comment:
ah I think I remember now if has empty contents if it has children.
would `if corporate.has_children?` not be better?